aiida/manage/tests/pytest_fixtures.py

```diff
@@ -0,0 +1,145 @@
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""
+Collection of pytest fixtures using the TestManager for easy testing of AiiDA plugins.
+
+ * aiida_profile
+ * clear_database
+ * aiida_localhost
+ * aiida_local_code_factory
+
+"""
+from __future__ import division
+from __future__ import absolute_import
+from __future__ import print_function
+
+import tempfile
+import shutil
+import pytest
+from aiida.manage.tests import test_manager, get_test_backend
+
+
+@pytest.fixture(scope='session', autouse=True)
+def aiida_profile():
+    """Set up AiiDA test profile for the duration of the tests.
+
+    Note: scope='session' limits this fixture to run once per session. Thanks to ``autouse=True``, you don't actually
+     need to depend on it explicitly - it will activate as soon as you import it in your ``conftest.py``.
+    """
+    # create new TestManager instance
+    with test_manager(backend=get_test_backend()) as test_mgr:
+        yield test_mgr
+    # here, the TestManager instance has already been destroyed
+
+
+@pytest.fixture(scope='function')
+def clear_database(aiida_profile):  # pylint: disable=redefined-outer-name
+    """Clear the database after each test.
+
+    Note: Thanks to ``autouse=True``, this happens as soon as you import this fixure in your ``conftest.py``.
+    """
+    yield
+    # after the test function has completed, reset the database
+    aiida_profile.reset_db()
+
+
+@pytest.fixture(scope='function')
+def temp_dir():
+    """Get a temporary directory.
+
+    E.g. to use as the working directory of an AiiDA computer.
+
+    :return: The path to the directory
+    :rtype: str
+
+    """
+    dirpath = tempfile.mkdtemp()
+    yield dirpath
+    # after the test function has completed, remove the directory again
+    shutil.rmtree(dirpath)
+
+
+@pytest.fixture(scope='function')
+def aiida_localhost(temp_dir):  # pylint: disable=redefined-outer-name
+    """Get an AiiDA computer for localhost.
+
+    Usage::
+
+      def test_1(aiida_localhost):
+          name = aiida_localhost.get_name()
+          # proceed to set up code or use 'aiida_local_code_factory' instead
+
+
+    :return: The computer node
+    :rtype: :py:class:`aiida.orm.Computer`
+    """
+    from aiida.orm import Computer
+    from aiida.common.exceptions import NotExistent
+
+    name = 'localhost'
+
+    try:
+        computer = Computer.objects.get(name=name)
+    except NotExistent:
+        computer = Computer(
+            name=name,
+            description='localhost computer set up by test manager',
+            hostname=name,
+            workdir=temp_dir,
+            transport_type='local',
+            scheduler_type='direct'
+        )
+        computer.store()
+        computer.configure()
```

Comment:
Should we really `configure` the computer as well already? There might be use cases where one does not want a configured computer yet and if one does, doing so is one line plus gives the opportunity of determining exactly how to configure

Reply:
Well, if you don't configure the computer, you can't use it, right?
Do you want to add another fixture that configures the unconfigured computer?

Reply:
In all my tests I never actually _use_ the computer, just like I don't use the `Code` instances. I just need those nodes as placeholders. I think that is a perfectly valid way of testing. In this case for me it does not matter if the computer _is_ configured, but I was just thinking to potential use cases where that would be a problem